File: Domain/Entities/Transport.cs
using Nirvachak_AI.Domain.Enums;

namespace Nirvachak_AI.Domain.Entities;

public class TransportVehicle
{
    public int Id { get; set; }
    public string DriverName { get; set; } = string.Empty;
    public string DriverPhone { get; set; } = string.Empty;
    public string? VehicleNumber { get; set; }
    public string? VehicleType { get; set; }   // Auto, Car, Van, Bus
    public int Capacity { get; set; }
    public int BoothNumber { get; set; }
    public int ConstituencyId { get; set; }
    public Constituency? Constituency { get; set; }
    public bool IsAvailable { get; set; } = true;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

public class VoterTransportRequest
{
    public int Id { get; set; }
    public int VoterId { get; set; }
    public Voter? Voter { get; set; }
    public int? VehicleId { get; set; }
    public TransportVehicle? Vehicle { get; set; }
    public TransportStatus Status { get; set; } = TransportStatus.Pending;
    public string? PickupAddress { get; set; }
    public string? PickupNotes { get; set; }
    public int ConstituencyId { get; set; }
    public Constituency? Constituency { get; set; }
    public string? RequestedByUserId { get; set; }
    public DateTime RequestedAt { get; set; } = DateTime.UtcNow;
    public DateTime? AssignedAt { get; set; }
    public DateTime? PickedUpAt { get; set; }
}

File: mobile/src/api/transport.ts
import apiClient from './client';

export interface TransportVehicleItem {
  id: number;
  driverName: string;
  driverPhone: string;
  vehicleNumber?: string;
  vehicleType?: string;
  capacity: number;
  boothNumber: number;
  isAvailable: boolean;
  notes?: string;
}

export interface TransportRequestItem {
  id: number;
  voterId: number;
  voterName: string;
  voterPhone?: string;
  vehicleId?: number;
  driverName?: string;
  vehicleNumber?: string;
  status: string;
  pickupAddress?: string;
  requestedAt: string;
}

export const VEHICLE_TYPES = ['Auto', 'Car', 'Van', 'Bus', 'Other'];
export const TRANSPORT_STATUSES = [
  { key: 'Pending',  label: 'Pending',    color: '#868e96' },
  { key: 'Assigned', label: 'Assigned',   color: '#3b5bdb' },
  { key: 'PickedUp', label: 'Picked Up',  color: '#f59f00' },
  { key: 'Voted',    label: 'Voted',      color: '#2f9e44' },
  { key: 'Cancelled',label: 'Cancelled',  color: '#e03131' },
];

export const getVehicles = async (): Promise<TransportVehicleItem[]> => {
  const { data } = await apiClient.get<TransportVehicleItem[]>('/transport/vehicles');
  return data;
};

export const createVehicle = async (req: {
  driverName: string; driverPhone: string; vehicleNumber?: string;
  vehicleType?: string; capacity: number; boothNumber: number; notes?: string;
}): Promise<void> => {
  await apiClient.post('/transport/vehicles', req);
};

export const getTransportRequests = async (): Promise<TransportRequestItem[]> => {
  const { data } = await apiClient.get<TransportRequestItem[]>('/transport/requests');
  return data;
};

export const createTransportRequest = async (req: {
  voterId: number; pickupAddress?: string; pickupNotes?: string; vehicleId?: number;
}): Promise<void> => {
  await apiClient.post('/transport/requests', req);
};

export const updateTransportStatus = async (id: number, status: string): Promise<void> => {
  await apiClient.put(`/transport/requests/${id}/status`, null, { params: { status } });
};
